### DISCUSSION TITLE
How can I find current Request info in my customized implementation of IntoResponse trait

### DISCUSSION THREAD

**Discussion by gy0801151351** [tokio-rs/axum#573] [Q&A] [answered]
I need to get the information of the Request struct in my customized implementation of IntoResponse trait.
For example, I need an extension info in current Request, and put some related info into my customized Response.
But I found the implemented method is "fn into_response(self) -> Response<Self::Body>", and I didn't know
how to find the Request info related with current Response because I don't want to build my customized Response by passing a Request parameter. I just want to use low coupling design.

**Comment by davidpdrsn** [comment, marked as answer]
Well the state needs to get passed from the request to the response somehow. You can either do that when constructing the response your handler or transferring it in a middleware. The docs have examples of how to write middleware.

There is no way to “magically” move data from requests to responses.
